src/index.ts

```diff
@@ -5,7 +5,7 @@ export const main: HttpFunction = async (req, res) => {
     res.set("Access-Control-Allow-Origin", "*")
 
     try {
-        if (!("barcode" in req.query)) {
+        if (!("barcode" in req.query || "uid" in req.query) || ("barcode" in req.query && "uid" in req.query)) {
```

Comment:
error message should be updated for the cases - simply doing "barcode no provided" for anything is a surefire way to go crazy when working with this cloud function